### PR TITLE
Change (future) archive download endpoint.

### LINKED ISSUE #3427: Fix /api/packages/<package>/versions/<version>.tar.gz
This seems to be an unused API call, because it is not working for a while now. The api generator parses the version pattern to include the `.tar.gz` postfix in the `version`, making version pattern check to fail.

We should:
- fix the API generator
- maybe remove this API if we are not going to use it?

### PR DESCRIPTION
**Pull request by isoos**
- Fixes #3427
- I think `[..]/archive.tar.gz` is better than `[..]/archive`, as it indicates that it will be a file in the given format.

### CODE REVIEW

File: app/lib/frontend/handlers/pubapi.dart [hunk 1]
```diff
@@ -45,7 +45,7 @@ class PubApi {
 
   /// Downloading package.
   /// https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md#download-a-specific-version-of-a-package
-  @EndPoint.get('/api/packages/<package>/versions/<version>.tar.gz')
+  @EndPoint.get('/api/packages/<package>/versions/<version>/archive.tar.gz')
   @EndPoint.get('/packages/<package>/versions/<version>.tar.gz')
```
Comment (jonasfj) [inline app/lib/frontend/handlers/pubapi.dart]: Can we add a comment here documenting that `/packages/<package>/versions/<version>.tar.gz` is hardcoded into the clients, so while this is deprecated we need to support it indefinitely?

**Comment by jonasfj** [comment]
Also we need to update the URLs in the JSON we return.

**Comment by jonasfj** [comment]
And write unit tests that the JSON we return contains URLs from which the archive can be fetcted..

**Comment by isoos** [comment]
> Also we need to update the URLs in the JSON we return.
> And write unit tests that the JSON we return contains URLs from which the archive can be fetcted..

Doing those in a subsequent PR.

**Pull request merged**